src/org/jetbrains/plugins/scala/compiler/ScalaHydraCompilerConfigurationPanel.scala

```diff
@@ -0,0 +1,81 @@
+package org.jetbrains.plugins.scala.compiler
+
+import java.awt.event.{ActionEvent, FocusEvent, FocusListener}
+import javax.swing.event.DocumentEvent
+
+import com.intellij.openapi.ui.Messages
+import org.jetbrains.plugins.scala.caches.HydraArtifactsCache
+import org.jetbrains.plugins.scala.extensions
+import com.intellij.openapi.project.Project
+import com.intellij.ui.DocumentAdapter
+import org.jetbrains.plugins.scala.project.{ProjectExt, Version, Versions}
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success}
+
+/**
+  * @author Maris Alexandru
+  */
+class ScalaHydraCompilerConfigurationPanel(project: Project, settings: HydraCompilerSettings) extends HydraCompilerConfigurationPanel {
+  private val UnknownVersion = "unknown"
+
+  val documentAdapter = new DocumentAdapter {
+      override def textChanged(documentEvent: DocumentEvent): Unit =
+        if (getUsername.nonEmpty && getPassword.nonEmpty) downloadButton.setEnabled(true)
+        else downloadButton.setEnabled(false)
+    }
+
+  val focusListener = new FocusListener {
+    override def focusGained(e: FocusEvent): Unit = {}
+
+    override def focusLost(e: FocusEvent): Unit = if (getUsername.nonEmpty && getPassword.nonEmpty &&
+      (HydraCredentialsManager.getLogin != getUsername || HydraCredentialsManager.getPlainPassword != getPassword)) {
+      HydraCredentialsManager.setCredentials(getUsername, getPassword)
+      hydraVersionComboBox.setItems(Versions.loadHydraVersions)
+    }
+  }
+
+  userTextField.addFocusListener(focusListener)
+  userTextField.getDocument.addDocumentListener(documentAdapter)
+  passwordTextField.getDocument.addDocumentListener(documentAdapter)
+  passwordTextField.addFocusListener(focusListener)
+  hydraVersionComboBox.setItems(Versions.loadHydraVersions)
+  downloadButton.addActionListener((_: ActionEvent) => onDownload())
+
+  def selectedVersion: String = hydraVersionComboBox.getSelectedItem.asInstanceOf[String]
```

Comment:
I wonder if it wouldn't be safer to use `.toString` instead of `asInstanceOf[String]`, but maybe that's how things are done in IJ?